packages/amplication-client/src/Workspaces/WorkspaceHeader/WorkspaceHeader.tsx

```diff
@@ -187,13 +260,27 @@ const WorkspaceHeader: React.FC = () => {
         <div className={`${CLASS_NAME}__center`}></div>
         <div className={`${CLASS_NAME}__right`}>
           <div className={`${CLASS_NAME}__links`}>
-            <Button
-              className={`${CLASS_NAME}__upgrade__btn`}
-              buttonStyle={EnumButtonStyle.Outline}
-              onClick={handleUpgradeClick}
-            >
-              Upgrade
-            </Button>
+            {upgradeButtonData.showUpgradeTrialButton && (
+              <ButtonProgress
+                className={`${CLASS_NAME}__upgrade__btn`}
+                onClick={handleUpgradeClick}
+                progress={upgradeButtonData.trialLeftProgress}
+                leftValue={`${upgradeButtonData.trialDaysLeft} days free trial left`}
+                yellowColorThreshold={50}
+                redColorThreshold={0}
+              >
+                Upgrade
+              </ButtonProgress>
+            )}
+            {upgradeButtonData.showUpgradeDefaultButton && (
```

Comment:
instead of having this condition I would add this to the upgrade button, as the controller of the button renderer